### PR TITLE
#282: Updated python to Python3.10

### LINKED ISSUE #282: AI-Lab: Update to Python 3.10
Initial changes
* [x] Update Minimum Python version in file `pyproject.toml`
* [x] Update dependencies `poetry update`

Update Operating system from ubuntu_20.04 to ubuntu_22.04 
* [x] for Docker Edition [exasol/ds/sandbox/lib/dss_docker/Dockerfile
](https://github.com/exasol/ai-lab/blob/main/exasol/ds/sandbox/lib/dss_docker/Dockerfile)  
* [x] for AMI [exasol/ds/sandbox/lib/config.py](https://github.com/exasol/ai-lab/blob/main/exasol/ds/sandbox/lib/config.py) (will be used for VM images, too)
* [x] Documentation
* [x] Tests [test/integration/test_create_dss_docker_image.py
](https://github.com/exasol/ai-lab/blob/main/test/integration/test_create_dss_docker_image.py)

More changes 
* [x] Update version numbers of apt packages in Ansible scripts
* [x] Update Matrix build definitions in files in `.github/workflows`

GitHub workflows 
* [x] Update actions/checkout to v4
* [x] Replace `./.github/actions/prepare_poetry_env` by `exasol/python-toolbox/.github/actions/python-environment@0.13.0`, see [SLC release](https://github.com/exasol/script-languages-release/blob/master/.github/workflows/check_version.yaml#L17)

### PR DESCRIPTION
**Pull request by tomuben**
fixes #282

### CODE REVIEW

File: exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt [hunk 1]
```diff
@@ -7,4 +7,4 @@ jupysql==0.10.10
 sqlalchemy_exasol==4.6.3
 stopwatch.py==2.0.1
 boto3==1.26.163
-exasol-notebook-connector @ git+https://github.com/exasol/notebook-connector.git@main
+exasol-notebook-connector @ git+https://github.com/exasol/notebook-connector.git@bug/tmp_fix_deps
```
Comment (ckunki) [inline exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt:10]: Do you plan to stay on this branch?

**Review by ckunki — CHANGES_REQUESTED** [review]
Please additionally update changes files.

**Review by ckunki — CHANGES_REQUESTED** [review]
Please additionally update the versions of the actions in the github workflows, eg. `actions/checkout`

File: exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt (hunk 1 above)
Comment (tomuben) [reply exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook_requirements.txt]: The other PRs in sagemaker-extension and transformer-extensions are now merged. I reverted this.

**Pull request merged**